compiler/rustc_mir/src/transform/promote_consts.rs

```diff
@@ -628,7 +687,22 @@ impl<'tcx> Validator<'_, 'tcx> {
                 }
 
                 match op {
-                    // FIXME: reject operations that can fail -- namely, division and modulo.
+                    BinOp::Div | BinOp::Rem => {
+                        if !self.explicit && lhs_ty.is_integral() {
+                            // Integer division: the RHS must be a non-zero const.
+                            let const_val = match rhs {
+                                Operand::Constant(c) => {
+                                    c.literal.try_eval_bits(self.tcx, self.param_env, lhs_ty)
```

Comment:
The one concern I have with this is that it could make promotion dependent on the value of a constant in another crate. I wonder if it is worth to somehow ensure that this is either a literal or a crate-local constant.

Reply:
I think it's fine. If the constant is evauable at this point, then it must be part of the public API of the other crate. Changing it can already cause breaking changes without taking promotion into account. The diagnostic is just a bit more suprising with promotion